server/channels/store/sqlstore/post_store.go

```diff
@@ -942,6 +942,9 @@ func (s *SqlPostStore) Delete(rctx request.CTX, postID string, time int64, delet
 		err = s.deleteThread(transaction, postID, time)
 	} else {
 		err = s.updateThreadAfterReplyDeletion(transaction, id.RootId, id.UserId)
+		if _, err = s.GetMasterX().Exec("UPDATE Posts SET UpdateAt = ? WHERE Id = ?", time, id.RootId); err != nil {
+			mlog.Warn("Error updating Post UpdateAt.", mlog.Err(err))
```

Comment:
Should be in the transaction, I think. And we should be using squirrel for new code.

Reply:
Added the operation to the transaction. Not sure if it has any potential implications if that particular part of the transaction fails, but probably is the best. That being said, when creating a post, a few things fall outside of the transaction, including updating the root post last post at, so not sure 100%.

Re: squirrel, I also updated it, but the code is not really "new". It was mostly a copy paste from what is used on save and update. I just considered creating a function like `updateThreadAfterReplyDeletion` to reuse in the three places it is used (create, update and delete) but the three of them handle transactions differently, so I guess such "unification" is better handled in a different PR. That being said, I am open to suggestions.